projects/workers/src/worker/classes/web-worker.ts

```diff
@@ -0,0 +1,78 @@
+import {fromEvent} from 'rxjs';
+import {takeWhile} from 'rxjs/operators';
+import {WorkerFunction} from '../types/worker-function';
+import {AnyNextSubject} from './any-next-subject';
+
+export class WebWorker<T = any, R = any> extends AnyNextSubject<R> {
+    private worker!: Worker;
+
+    constructor(private url: string, options?: WorkerOptions) {
+        super();
+
+        try {
+            this.worker = new Worker(url, options);
+        } catch (e) {
+            this.error(e);
+        }
+
+        fromEvent<MessageEvent>(this.worker, 'message')
+            .pipe(takeWhile(() => !this.isStopped))
+            .subscribe(event => {
+                if (event.data) {
+                    if (event.data.hasOwnProperty('error')) {
+                        this.error(event.data.error);
+                    } else if (event.data.hasOwnProperty('result')) {
+                        super.next(event.data.result);
+                    } else {
+                        super.next();
+                    }
+                }
+            });
+
+        fromEvent<ErrorEvent>(this.worker, 'error')
+            .pipe(takeWhile(() => !this.isStopped))
+            .subscribe(event => {
+                this.error(event.error);
+            });
+    }
+
+    public static fromFunction<T, R>(
+        fn: WorkerFunction<T, R>,
+        options?: WorkerOptions,
+    ): WebWorker<T, R> {
+        return new WebWorker<T, R>(WebWorker.createFnUrl(fn), options);
+    }
+
+    private static createFnUrl(fn: WorkerFunction): string {
+        const script = `
+self.addEventListener('message', function(e) {
```

Comment:
let's write it as a valid JS code and then use toString() because template string with a code is very hard to read

Reply:
I can't. Well, I can use toString (), but then the coverage process breaks. It mutates the source code and adds constructs that are not available during the worker process.